### DISCUSSION TITLE
Is Imagemagick slow?

### DISCUSSION THREAD

**Discussion by rquadling** [ImageMagick/ImageMagick#2170] [General] [answered]
Before answering I want to make the following statement.

I have no axe to grind. I represent no one. I was just REALLY REALLY shocked with a comparison to Sharp published on Sharp's site (so yes .... pinch of salt ... etc. but all the same not a little faster but several factors faster!)

So. My goal is to do some image manipulation in an AWS Lambda function. Nothing super complicated.

To that end I came across https://github.com/serverlesspub/imagemagick-aws-lambda-2

It does what I need (in terms of reading what it does). The example is great, covering most of my use case. I can proceed.

Then one of my colleagues suggested using Sharp. I said "Hmm, isn't that a JS lib?". "Yes", he said. "What about performance?", I said. "Look at these stats!", he said and pointed me to https://sharp.pixelplumbing.com/performance

And seeing such a big difference in performance has led me here.

I'm not looking for a fight, but is there anyone who can reasonably explain how these differences are so great between ImageMagick and Sharp?

**Comment by urban-warrior** [comment, marked as answer]
See https://imagemagick.org/script/architecture.php.  ImageMagick is a general purpose image processing package and focus is on high quality results rather than performance.  We do address some performance issues by running algorithms across all the processors of a host in parallel and in some cases, we can leverage the host's GPU.  We did not review the benchmark.  However, notice that the benchmark uses the ImageMagick NPM package which we did not write and may not be performance optimized.  If you run the benchmark, add `-define jpeg:size=720x588" to the command-line before the JPEG image is read.  You can expect a substantial speed-up as illustrated here:

```
$ time convert logo.jpg -resize 720x588 im.jpg
0.604u 0.072s 0:00.17 394.1%    0+0k 0+72io 0pf+0w
$ time convert -define jpeg:size=720x588 logo.jpg -resize 720x588 im.jpg
0.225u 0.052s 0:00.09 300.0%    0+0k 0+72io 0pf+0w
```

**Comment by rquadling** [comment]
Thank you.
